### ISSUE TITLE
Cypress Command Failing - cy.get()

### ISSUE THREAD

**Issue by jsfinesse** [shivamguys/irctc-cypress-automation#71]
**Describe the bug**
When the automation lands on the page - https://www.irctc.co.in/nget/booking/train-list, it calls the command -
```
cy.get(
`:nth-child(${index + 2}) > .bull-back > app-train-avl-enq > :nth-child(1) > :nth-child(7) > :nth-child(1)`)
```

Whichs ends in the error - ```AssertionError: Timed out retrying after 120000ms: Expected to find element: `:nth-child(2) > .bull-back > app-train-avl-enq > :nth-child(1) > :nth-child(7) > :nth-child(1)`, but never found it.```

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 129.0.6668.70 (Official Build) (64-bit)

**Comment by jsfinesse** [comment]
At the page with the element visible but cy.get() is running unable to find it

![image](https://github.com/user-attachments/assets/43e4be17-e5c3-4045-9e97-fa5588af203e)

**Comment by 1CaptainPeroxide** [comment]
Same error here

**Comment by jsfinesse** [comment]
I have added a PR with changes that fix this issue, @shivamguys can review it and check it is a valid fix.

Meanwhile @1CaptainPeroxide and others can use my fork's changes.

**Comment by 1CaptainPeroxide** [comment]
> I have added a PR with changes that fix this issue, @shivamguys can review it and check it is a valid fix.
> 
> Meanwhile @1CaptainPeroxide and others can use my fork's changes.

Its working fine

**Comment by shivamguys** [comment]
Thanks for looking into it, really appreciate your efforts, have merged the changes 👍 @jsfinesse

**Comment by shivamguys** [comment]
By the way just asking have you guys booked some tatkal tickets with this @jsfinesse @1CaptainPeroxide

**Comment by 1CaptainPeroxide** [comment]
> By the way just asking have you guys booked some tatkal tickets with this @jsfinesse @1CaptainPeroxide
> 
> If yes could you tell is it worthwhile or not

Its working fine, i have used it for normal tickets several times, I didn't need tatkal tickets till now, but it is able to book normal tickets in less than 30 seconds.

**Comment by 1CaptainPeroxide** [comment]
The captcha part is just taking more time, otherwise its very fast. What can i do to improve captcha speed or i can use manual captcha?

**Comment by shivamguys** [comment]
I had once integrated google cloud vision apis for captchas processing that was very fast I am thinking 🤔 to re integrate that back again

**Comment by kumarsonungp** [comment]
> I had once integrated google cloud vision apis for captchas processing that was very fast I am thinking 🤔 to re integrate that back again

i'll be waiting for this integration.

**Comment by jsfinesse** [comment]
> By the way just asking have you guys booked some tatkal tickets with this @jsfinesse @1CaptainPeroxide
> 
> If yes could you tell is it worthwhile or not

Yes, I tried booking today; it works, but I could book successfully only on the second try. The first try failed at the payment selection page due to IRCTC's request not being processed. That might have been caused by dynamic fare as I had opted for Premium Tatkal.

The major pain point is captcha solving. Its failure rate is quite high.

Thanks for the project, though, it really helped me out today.
